.github/workflows/release.yml

```diff
@@ -0,0 +1,163 @@
+            manylinux: true
+          - platform: macos-x86_64
+            os: macos-12
+            # By default we have to target macOS 10.14 at minimum according to CRAN policy.
+            # See https://stat.ethz.ch/pipermail/r-package-devel/2023q4/010078.html
+            MACOSX_DEPLOYMENT_TARGET: 10.14
+            triplet: x64-osx-release
+          - platform: macos-arm64
+            os: macos-12
+            cmake_args: -DCMAKE_OSX_ARCHITECTURES=arm64
+            MACOSX_DEPLOYMENT_TARGET: 11
+            triplet: arm64-osx-release
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.manylinux && 'quay.io/pypa/manylinux2014_x86_64:2023-11-13-f6b0c51' || '' }}
+    env:
+      MACOSX_DEPLOYMENT_TARGET: ${{ matrix.MACOSX_DEPLOYMENT_TARGET }}
+      VCPKG_BINARY_SOURCES: 'clear;x-gha,readwrite'
+
+    steps:
+      - name: Checkout TileDB
+        # v4 uses node 20 which is incompatible with the libc version of the manylinux image
+        uses: actions/checkout@v3
+      - name: Export GitHub Actions cache variables
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.exportVariable('ACTIONS_CACHE_URL', process.env.ACTIONS_CACHE_URL || '');
+            core.exportVariable('ACTIONS_RUNTIME_TOKEN', process.env.ACTIONS_RUNTIME_TOKEN || '');
+      - name: Set variables
+        id: get-values
+        run: |
+          release_hash=$( echo ${{ github.sha }} | cut -c-8 - )
+          ref=${{ github.head_ref || github.ref_name }}
+          echo "release_hash=$release_hash" >> $GITHUB_OUTPUT
+          echo "archive_name=tiledb-${{ matrix.platform }}-${ref##*/}-$release_hash" >> $GITHUB_OUTPUT
+        shell: bash
+      - name: Install manylinux prerequisites
+        if: matrix.manylinux
+        run: |
+          set -e pipefail
+          yum install -y redhat-lsb-core centos-release-scl devtoolset-7
+          echo "source /opt/rh/devtoolset-7/enable" >> ~/.bashrc
+      - name: Configure TileDB
+        run: |
+          cmake -S . -B build \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_INSTALL_PREFIX=./dist \
+            -DTILEDB_S3=ON \
+            -DTILEDB_AZURE=ON \
+            -DTILEDB_GCS=ON \
+            -DTILEDB_HDFS=${{ startsWith(matrix.platform, 'windows') && 'OFF' || 'ON' }} \
+            -DTILEDB_SERIALIZATION=ON \
+            -DTILEDB_WEBP=ON \
+            -DTILEDB_STATIC=OFF \
+            -DTILEDB_TESTS=OFF \
+            -DVCPKG_TARGET_TRIPLET=${{ matrix.triplet }} \
+            ${{ matrix.cmake_args }}
+        shell: bash
+      - name: Build TileDB
+        run: cmake --build build --config Release
+      - name: Install TileDB
+        run: cmake --build build --config Release --target install-tiledb
+      - name: Archive installed artifacts (non-Windows)
+        if: ${{ !startsWith(matrix.platform, 'windows') }}
+        run: |
+          tar -czf ${{ steps.get-values.outputs.archive_name }}.tar.gz -C dist .
```

Comment:
As a follow-up (soon) we should turn this into a target -- it is very useful to be able to run `make release` and get usable packages.